After SpotBugs Maven Plugin upgrade to version 4.9.5.0 there is a false/positive RANGE_ARRAY_INDEX finding

Since updating to version 4.9.5.0 (before: 4.9.4.2) for the Spotbugs Maven Plugin, we receive the following error for the folowing Java code line:
```
  public @Nullable String provideTransactionName(@NotNull HttpServletRequest request) {
    return String.format("%s %s", request.getMethod(), request.getRequestURI());
  }
```

leads to error:
```
[ERROR] Medium: Array index is out of bounds: 2 [...] At XYZ.java:[line 21]At XYZ.java:[line 21] RANGE_ARRAY_INDEX
```

I can't find any error in this line, also it is a "no brainer" ;)


Thank you in advance!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,8 @@ This is the changelog for SpotBugs. This follows [Keep a Changelog v1.0.0](http:
 Currently the versioning policy of this project follows [Semantic Versioning v2.0.0](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased - 2025-??-??
+### Fixed
+- Fix exception throw when analyzing `jakarta.servlet.http.HttpServletRequest` method calls ([#3711](https://github.com/spotbugs/spotbugs/issues/3711))
 
 ## 4.9.5 - 2025-09-14
 ### Fixed

--- a/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue3711Test.java
+++ b/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue3711Test.java
@@ -1,0 +1,13 @@
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.AbstractIntegrationTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class Issue3711Test extends AbstractIntegrationTest {
+
+    @Test
+    void test() {
+        Assertions.assertDoesNotThrow(() -> performAnalysis("ghIssues/Issue3711.class"));
+    }
+}

--- a/spotbugs/src/main/java/edu/umd/cs/findbugs/OpcodeStack.java
+++ b/spotbugs/src/main/java/edu/umd/cs/findbugs/OpcodeStack.java
@@ -2631,7 +2631,7 @@ public class OpcodeStack {
             }
         } else if (seen == Const.INVOKEINTERFACE && "getParameter".equals(method)
                 && ("javax/servlet/http/HttpServletRequest".equals(clsName) || "javax/servlet/http/ServletRequest".equals(clsName)
-                || "jakarta/servlet/http/HttpServletRequest".equals(clsName) || "jakarta/servlet/http/ServletRequest".equals(clsName))) {
+                        || "jakarta/servlet/http/HttpServletRequest".equals(clsName) || "jakarta/servlet/http/ServletRequest".equals(clsName))) {
             Item requestParameter = pop();
             pop();
             Item result = new Item("Ljava/lang/String;");

--- a/spotbugs/src/main/java/edu/umd/cs/findbugs/OpcodeStack.java
+++ b/spotbugs/src/main/java/edu/umd/cs/findbugs/OpcodeStack.java
@@ -2630,8 +2630,8 @@ public class OpcodeStack {
                 return;
             }
         } else if (seen == Const.INVOKEINTERFACE && "getParameter".equals(method)
-                && ("javax/servlet/http/HttpServletRequest".equals(clsName) || "javax/servlet/http/ServletRequest".equals(clsName))
-                || "jakarta/servlet/http/HttpServletRequest".equals(clsName) || "jakarta/servlet/http/ServletRequest".equals(clsName)) {
+                && ("javax/servlet/http/HttpServletRequest".equals(clsName) || "javax/servlet/http/ServletRequest".equals(clsName)
+                || "jakarta/servlet/http/HttpServletRequest".equals(clsName) || "jakarta/servlet/http/ServletRequest".equals(clsName))) {
             Item requestParameter = pop();
             pop();
             Item result = new Item("Ljava/lang/String;");

--- a/spotbugsTestCases/src/java/ghIssues/Issue3711.java
+++ b/spotbugsTestCases/src/java/ghIssues/Issue3711.java
@@ -1,0 +1,9 @@
+package ghIssues;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+public class Issue3711 {
+    public String provideTransactionName(HttpServletRequest request) {
+        return String.format("%s %s", request.getMethod(), request.getRequestURI());
+    }
+}

